flocker/node/functional/test_deploy.py

```diff
@@ -360,11 +360,13 @@ def inspect_application(_):
             deferred_list = docker_client.list()
 
             def app_memory(units):
-                unit = units.pop()
-                self.assertEqual(unit.mem_limit, EXPECTED_MEMORY_LIMIT)
-                return deferred_list
-
+                checked = [
+                    self.assertEqual(unit.mem_limit, EXPECTED_MEMORY_LIMIT)
+                    for unit in units if unit.name == application_name
+                ]
```

Comment:
It's sort of weird to put the `assertEqual` into this expression.  Instead, I'd collect units with a matching name and make the assertion afterwards.